sympy/functions/special/bessel.py

```diff
@@ -1710,20 +1722,28 @@ class marcumq(Function):
     @classmethod
     def eval(cls, m, a, b):
         from sympy import exp, uppergamma
-        if a == 0:
-            if m == 0 and b == 0:
+        if a is S.Zero:
+            if m is S.Zero and b is S.Zero:
```

Comment:
I see why you were asking about sympification of arguments to `eval`. I think it's fine to treat `eval` as an internal routine so that sympification is the user's responsibility.